app/src/main/kotlin/com/wire/android/core/usecase/UseCaseExecutor.kt

```diff
@@ -0,0 +1,29 @@
+package com.wire.android.core.usecase
+
+import com.wire.android.core.exception.Failure
+import com.wire.android.core.functional.Either
+import kotlinx.coroutines.*
+
+interface UseCaseExecutor {
+    operator fun <T, P> UseCase<T, P>.invoke(
+        scope: CoroutineScope,
```

Comment:
Are we applying here the same style (regarding function arguments) as we agreed on the other previous PRs?

Reply:
Well, 
```
operator fun <T, P> UseCase<T, P>.invoke(scope: CoroutineScope, params: P, dispatcher: CoroutineDispatcher = Dispatchers.IO, onResult: (Either<Failure, T>) -> Unit = {})
```
this doesn't fit in one line.


```
operator fun <T, P> UseCase<T, P>.invoke(
scope: CoroutineScope, params: P, dispatcher: CoroutineDispatcher = Dispatchers.IO, onResult: (Either<Failure, T>) -> Unit = {}
)
```
this does, but it looks very disorganized compared to what we have now. 

Since this is a method declaration, I would like to keep it as it is for the sake of readability. Especially with default values etc, it becomes harder to read. However, when using this method we can of course list all the parameters in one line:
```
myUseCase.invoke(viewModelScope, UseCaseParams(), Dispatchers.Default, ::onResult)
```